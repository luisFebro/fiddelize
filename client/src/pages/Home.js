import React, { Fragment } from 'react';
import RegisterClientAdmin from '../components/auth/RegisterClientAdmin';
import ScrollArrow from '../keyframes/built/scroll-arrow/ScrollArrow';
import { Link } from 'react-router-dom';
import AOS from 'aos';
// import ModalForPermission from '../components/pwa-push-notification/ModalForPermission';
// import showVanillaToast from '../components/vanilla-js/toastify/showVanillaToast'

const isSmall = window.Helper.isSmallScreen();

export default function Home() {
    AOS.init();
    // const showToast = () => {
    //     showVanillaToast("Testing", 10000, {avatar: ' '});
    // }

    const showSlogon = () => (
        <section className="mt-5 d-flex flex-column flex-md-row justify-content-center align-items-center">
            <div className="align-self-center m-3">
                <img
                    className="svg-elevation"
                    src="/img/official-logo-white.png"
                    alt="logo"
                    width={150}
                    height='auto'
                />
            </div>
            <h1
                className="text-center align-item-center text-md-left text-title text-white"
                style={{ maxWidth: '700px'}}
            >
                Pontos de Fidelidade Digital para seus Clientes
            </h1>
        </section>
    );

    const showAppShowCase = () => (
        <section>
            <p
                data-aos="fade-up"
                data-aos-duration="1500"
                className="ml-4 text-title text-white"
            >
                Combinamos design, sistema e tecnologia de ponta para entregar
                uma experiência única para empreendedores e seus clientes através
                de nossa plataforma de fidelização.
            </p>
            <div style={{margin: isSmall ? '10px 0 100px 0' : '50px 0 100px 0'}} className="d-flex justify-content-center">
                <ScrollArrow color="white" />
            </div>
            <p
                data-aos="fade-up"
                data-aos-duration="1500"
                className="ml-4 text-title text-white text-center"
            >
                Crie hoje seu multi App de<br />
                pontos de fidelidade com<br />
                tecnologia sob medida movido a<br />
                aumentar sua clientela e vendas<br />
            </p>
            <div
                data-aos="fade-up-right"
                data-aos-duration="1500"
                style={{maxWidth: 800, position: 'relative', left: isSmall ? '-100px' : '-230px'}}
            >
                <img className="img-fluid shape-elevation" src="/img/illustrations/one-hand-held-mobile.png" height="auto" alt="app do celular"/>
            </div>
        </section>
    );

    return(
        <Fragment>
            <span className="text-right text-white for-version-test">
                {"T62"}
            </span>
            {showSlogon()}
            <div style={{margin: isSmall ? '10px 0 100px 0' : '50px 0 100px 0'}} className="d-flex justify-content-center">
                <ScrollArrow color="white" />
            </div>
            {showAppShowCase()}
            <RegisterClientAdmin />
        </Fragment>
    );
};


/* ARCHIVES
<div className="mt-3 text-subtitle text-center">Acumule pontos e ganhe produtos e serviços</div>
<Link to="/regulamento">
    <div
        className="my-5 text-subtitle font-weight-italic text-center"
        style={{color: "var(--mainPink)", cursor: "pointer"}}
    >
        Consulte<br />as Regras Aqui
    </div>
</Link>
*/