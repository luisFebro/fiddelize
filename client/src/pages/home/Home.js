import React, { Fragment, useState } from 'react';
import VAsyncRegisterClientAdmin from '../../components/auth/VAsyncRegisterClientAdmin';
import ScrollArrow from '../../keyframes/built/scroll-arrow/ScrollArrow';
import { Link } from 'react-router-dom';
import AppShowCase from './AppShowCase';
import { CLIENT_URL } from '../../config/clientUrl';
import CompLoader from '../../components/CompLoader';

export default function Home() {
    const [viewPhone, setViewPhone] = useState(false);

    const isSmall = React.useCallback(window.Helper.isSmallScreen(), []);

    const showSlogon = () => (
        <section className="mt-3 d-flex flex-column flex-md-row justify-content-center align-items-center">
            <div className="align-self-center m-1">
                <picture>
                    <source srcSet="/img/official-logo-white.webp" type="image/webp" />
                    <source srcSet="/img/official-logo-white.png" type="image/png" />
                    <img
                        className="svg-elevation"
                        src={`${CLIENT_URL}/img/official-logo-white.webp`}
                        alt="logo"
                        width={150}
                        height='auto'
                    />
                </picture>
            </div>
            <h1
                className="mx-2 mx-md-5 text-center align-item-center text-md-left text-title text-white"
                style={{ maxWidth: '700px'}}
            >
                Pontos de Fidelidade Digital para seus Clientes
            </h1>
        </section>
    );

    return(
        <Fragment>
            <section className="full-height">
                <span className="text-right text-white for-version-test">
                    {"T99"}
                </span>
                {showSlogon()}
                <div style={{margin: isSmall ? '10px 0 100px 0' : '50px 0 100px 0'}} className="d-flex justify-content-center">
                    <ScrollArrow color="white" />
                </div>
            </section>
            <AppShowCase setData={setViewPhone} />
            <CompLoader
                comp={<VAsyncRegisterClientAdmin />}
                hide={viewPhone}
                timeout={3500}
                width={200}
                marginY={30}
            />
        </Fragment>
    );
};


/* ARCHIVES

<div className="mt-3 text-subtitle text-center">Acumule pontos e ganhe produtos e servi√ßos</div>
<Link to="/regulamento">
    <div
        className="my-5 text-subtitle font-weight-italic text-center"
        style={{color: "var(--mainPink)", cursor: "pointer"}}
    >
        Consulte<br />as Regras Aqui
    </div>
</Link>
*/
