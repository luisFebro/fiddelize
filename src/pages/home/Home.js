import { useEffect } from "react";
import { setVar } from "init/var";
import { CLIENT_URL } from "config/clientUrl";
import "./_Home.scss";
import FeaturesSection from "./sections/features/FeaturesSection";
import CtaSection from "./sections/CtaSection";
import preloadMedia from "utils/media/preloadMedia";
// import CompLoader from "../../components/CompLoader";
// import AppShowCase from "./AppShowCase";
// import { Link } from 'react-router-dom';

export default function Home({ match }) {
    useEffect(() => {
        preloadMedia({
            href: `${CLIENT_URL}/img/illustrations/home/feature-mobiles/cli-admin.gif`,
        });
        preloadMedia({
            href: `${CLIENT_URL}/img/illustrations/home/feature-mobiles/cli-member.gif`,
        });
        preloadMedia({
            href: `${CLIENT_URL}/img/illustrations/home/feature-mobiles/cli-user.gif`,
        });

        const gotReferrer = match.params.associateId;
        if (gotReferrer) {
            setVar({ referrer: gotReferrer }, "pre_register");
        }
    }, [match]);
    // const [viewPhone, setViewPhone] = useState(false);
    const showHero = () => (
        <section className="hero-section">
            <div className="container">
                <section className="row justify-content-center">
                    <div className="col-lg-45 order-lg-2">
                        <div className="hero-image">
                            <img
                                src="/img/illustrations/home/3d-illustra/hero-image-fiddelize.png"
                                alt="Criamos motivos para clientes comprarem do seu negócio."
                            />
                        </div>
                    </div>
                    <div className="col-lg-55 order-lg-1">
                        <div className="hero-content">
                            <h1 className="text-title line-height-hero">
                                Criamos mais motivos para os clientes comprarem
                                do seu negócio.
                            </h1>
                            <p className="mt-4 text-white">
                                Faça seu app hoje. Ofereça aos seus clientes{" "}
                                <strong>pontos</strong> como moeda digital para
                                trocarem por benefícios com{" "}
                                <strong>jogos de compra</strong>.
                            </p>
                        </div>
                    </div>
                </section>
            </div>
        </section>
    );

    return (
        <section className="home-page--root">
            {showHero()}
            <FeaturesSection />
            <CtaSection />
        </section>
    );
}
