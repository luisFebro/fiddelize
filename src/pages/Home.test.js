// 90% of time, you will use shallow. this tests one component at time. If there is another comp in the rendering, it will be ignored.
// mount for DOM browser rendering
// TIP: if the component has many functionalities such redux and diffent functions, try to write a new component with only the essencial logic...
// render
import { shallow } from "enzyme";
import Home from "./Home";

it("expect to render Home Component", () => {
    expect(shallow(<Home />)).toMatchSnapshot(); // n1
});

// it("expect to render Home Component", () => {
//     expect(shallow(<Home />).length).toEqual(1);
// })
// console.log(shallow(<Home />));

/* COMMENTS
n1: Snapshot tests are a very useful tool whenever you want to make sure your UI does not change unexpectedly.
A typical snapshot test case for a mobile app renders a UI component, takes a snapshot, then compares it to a reference snapshot file stored alongside the test. The test will fail if the two snapshots do not match: either the change is unexpected, or the reference snapshot needs to be updated to the new version of the UI component.
https://jestjs.io/docs/en/snapshot-testing

npm test -- --coverage = to see all components testing coverage.
*/
