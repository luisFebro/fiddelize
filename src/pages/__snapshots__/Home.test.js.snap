// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Home Component 1`] = `ShallowWrapper {}`;
